app/models/post.rb

```diff
@@ -1,3 +1,20 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :user, foreign_key: 'author_id'
+  has_many :likes
+  has_many :comments
+
+  after_create :increment_post_counter
+  after_destroy :decrement_post_counter
+
+  def increment_post_counter
+    user.increment!(:posts_counter)
```

Comment:
[REQUIRED] Kindly change this to `author.increment` since the author is supposed to be available to the post model in this scenario

Reply:
Done! thanks!